File: 0392_Is Subsequence/IsSubsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        if(string.IsNullOrEmpty(s)) return true;
        if(string.IsNullOrEmpty(t)) return false;
        
        var n = t.Length;
        var index = 0;
        
        foreach(var c in s)
        {
            while(index < n && c != t[index]) index++;
            if(index++ == n) return false;
        }

        return true;
    }
}

File: 0392_Is Subsequence/IsSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int i = 0;
        for(char c : t.toCharArray()) {
            if(i == s.length()) break;
            if(s.charAt(i) == c) i++;
        }

        return i == s.length();
    }
}